### PR TITLE
Documentation of environments API

### PR DESCRIPTION
**Pull request by GaneshSPatil**

### CODE REVIEW

File: source/includes/environment_config/_00-describe.md.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<%=
+describe_object 'The environment config object', since: '16.6.0' do
+  name                      'String',  'The environment config name'
```
Comment (varshavaradarajan) [inline source/includes/environment_config/_00-describe.md.erb]: Please change this to 16.7 :)

Comment (varshavaradarajan) [inline source/includes/environment_config/_00-describe.md.erb]: What about having 'The name of the environment'

File: source/includes/environment_config/_00-describe.md.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  pipelines                 'Array',   'The list of pipeline names that belongs to the specified environment'
+  agents                    'Array',   'The list of agent uuids that belongs to the specified environment'
+  environment_variables     'Array',   'The list of [environment variables](#environment-config-environment-variables) that will be passed to all tasks (commands) that are part of this environment.'
```
Comment (varshavaradarajan) [inline source/includes/environment_config/_00-describe.md.erb]: Is there any difference between ```pipeline-config-environment-variables``` and ```environment-config-environment-variables``` ? You can probably have a link to the ```pipeline-config-environment-variables``` and delete ```_01-environment-variable.md.erb``` file

File: source/includes/environment_config/_10-index.md.erb [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      }
+    ]
+  }
+}
+```
+
+Lists all available environments.
+
+<%= available_since('16.6.0') %>
+
```
Comment (varshavaradarajan) [inline source/includes/environment_config/_10-index.md.erb]: Please change this to 16.7.0

File: source/includes/environment_config/_20-show.md.erb [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+X-Content-Type-Options: nosniff
+X-UA-Compatible: chrome=1
+X-Controller-Action: api_v1/admin/environments#show
+Content-Type: application/vnd.go.cd.v1+json; charset=utf-8
+Cache-Control: max-age=0, private, must-revalidate
+X-Request-Id: 2ba62881-9778-4422-9947-c5d9c100f821
+X-Runtime: 0.166000
+Vary: Accept-Encoding, User-Agent
+ETag: "4167e3ec81fdac0fb29d854b36ceb981"
+Transfer-Encoding: chunked
```
Comment (ketan) [inline source/includes/environment_config/_20-show.md.erb]: Please not so many headers. Just need the etag and http header.

File: source/includes/environment_config/_30-create.md.erb [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+```json
+{
+  "_links": {
+    "self": {
+      "href": "https://ci.example.com/go/api/admin/environments/new_environment"
+    },
+    "doc": {
+      "href": "http://api.go.cd/#environment_config"
```
Comment (ketan) [inline source/includes/environment_config/_30-create.md.erb]: no underscores in url fragments. None of the existing urls have underscores in them. Please fix the representer and this

File: source/includes/environment_config/_30-create.md.erb [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  },
+  "name": "new_environment",
+  "pipelines": [
+    {
+      "_links": {
+        "self": {
+          "href": "https://ci.example.com/go/api/admin/pipelines/up42"
+        },
+        "doc": {
+          "href": "http://api.go.cd/#pipeline_config"
```
Comment (ketan) [inline source/includes/environment_config/_30-create.md.erb]: Please fix the pipeline config representer to use `#pipeline-config`

File: source/includes/environment_config/_40-update.md.erb [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+Atleast one of the following properties must be specified, not specifying the property will leave it unchanged.
+
+<%=
+  describe_object nil do
+    name                      'String',  'The environment config name'
+  pipelines                 'Array',   'The list of pipeline names that belongs to the specified environment'
+  agents                    'Array',   'The list of agent uuids that belongs to the specified environment'
+  environment_variables     'Array',   'The list of [environment variables](#environment-config-environment-variables) that will be passed to all tasks (commands) that are part of this environment.'
+  end
```
Comment (ketan) [inline source/includes/environment_config/_40-update.md.erb]: Please indent code properly.

File: source/includes/environment_config/_50-delete.md.erb [hunk 8]
```diff
@@ -0,0 +1,33 @@
+## Delete an environment
+
+```shell
+$ curl 'http://localhost:8153/go/api/admin/environments/my_environment' \
```
Comment (ketan) [inline source/includes/environment_config/_50-delete.md.erb]: Did you mean to say `ci.example.com`?

File: source/includes/environment_config/_30-create.md.erb [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+}
+```
+
+Creates an environment
+
+<%= available_since('16.6.0') %>
+
+<p class='http-request-heading'>HTTP Request</p>
+
+`POST /go/api/admin/environments/`
```
Comment (ketan) [inline source/includes/environment_config/_30-create.md.erb]: No trailing slash

File: source/includes/environment_config/_30-create.md.erb [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+The following properties may be specified:
+
+<%=
+  describe_object nil do
+  name                      'String',  'The environment config name'
+  pipelines                 'Array',   'The list of pipeline names that belongs to the specified environment'
+  agents                    'Array',   'The list of agent uuids that belongs to the specified environment'
```
Comment (ketan) [inline source/includes/environment_config/_30-create.md.erb]: how about "List of pipeline names/agent uuids that should be added to this environment"?

File: source/includes/environment_config/_40-update.md.erb [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        pipelines : [ { "name" : "pipeline1" } ]
+      }'
+
+```
+
+> The above command returns JSON structured like this:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/vnd.go.cd.v1+json; charset=utf-8
```
Comment (ketan) [inline source/includes/environment_config/_40-update.md.erb]: Please add the `ETag` header here too

File: source/includes/environment_config/_10-index.md.erb [hunk 12]
```diff
@@ -0,0 +1,116 @@
+## Get all environments
+
+```shell
+$ curl 'https://ci.example.com/go/api/admin/environments' \
+      -u 'username:password' \
+      -H 'Accept: application/vnd.go.cd.v1+json' \
+      -i
```
Comment (ketan) [inline source/includes/environment_config/_10-index.md.erb]: Omit the `-i`

File: source/includes/environment_config/_20-show.md.erb [hunk 13]
```diff
@@ -0,0 +1,109 @@
+## Get environment config
+
+```shell
+$ curl 'https://ci.example.com/go/api/admin/environments/my_environment' \
+      -u 'username:password' \
+      -H 'Accept: application/vnd.go.cd.v1+json' \
+      -i
```
Comment (ketan) [inline source/includes/environment_config/_20-show.md.erb]: Omit the `-i`

**Pull request merged**